src/Services/Order/OrderCreateService.php

```diff
@@ -454,15 +459,51 @@ private function getVariationById(int $variationId)
 	/**
 	 * Create order comments.
 	 *
-	 * @param int   $orderId
+	 * @param Order $order
 	 * @param array $data
 	 */
-	private function createOrderComments(int $orderId, array $data)
+	private function createOrderComments(Order $order, array $data)
 	{
+	    $orderId = $order->id;
+
 		try {
 			/** @var CommentRepositoryContract $commentRepo */
 			$commentRepo = pluginApp(CommentRepositoryContract::class);
 
+			/** @var OrderItemPropertyRepositoryContract $orderItemPropertyRepo */
+			$orderItemPropertyRepo = pluginApp(OrderItemPropertyRepositoryContract::class);
+
+            //save personalization information
+            foreach ($data['Transactions'] as $transaction) {
+                if (isset($transaction['variations'])) {
+                    foreach ($transaction['variations'] as $attribute) {
+                        if (in_array($attribute['formatted_name'], self::PERSONALIZATION_NAMING)) {
+                            foreach ($order->orderItems as $orderItem) {
+                                try {
+                                    $orderItemExternalItemIdProperty = $orderItemPropertyRepo->findByOrderItemIdAndTypeId(
+                                        $orderItem->id,
+                                        OrderPropertyType::EXTERNAL_ITEM_ID
+                                    );
+                                    $orderItemExternalItemId = $orderItemExternalItemIdProperty->value;
+                                } catch (\Exception $e) {
+                                    continue;
+                                }
```

Comment:
```suggestion
                                $orderItemExternalItemId = $orderItem->property(OrderPropertyType::EXTERNAL_ITEM_ID);
```

this function returns the value if property exists or null if not.